fsspec/spec.py

```diff
@@ -1437,14 +1437,28 @@ def to_dict(self) -> Dict[str, Any]:
         multiple), ``args`` (positional args, usually empty), and all other
         keyword arguments as their own keys.
         """
+        from .json import FilesystemJSONEncoder
+
+        json_encoder = FilesystemJSONEncoder()
+
+        def serialize(obj: Any) -> Any:
+            if isinstance(obj, (str, int, float, bool)):
+                return obj
+            if isinstance(obj, dict):
```

Comment:
consider `collections.abc.Mapping` and `collections.abc.Iterable`? Note that the latter also catches strings, which would then need to be tested for first.

Reply:
Sure!